### PR TITLE
Fix errors in stubs when running in strict mode

### PR DESCRIPTION
**Pull request by FuegoFro**
Add types to tempfile parameters, have subTest actually return a context manager.

### CODE REVIEW

File: stdlib/3/tempfile.pyi [hunk 1]
```diff
@@ -20,21 +21,19 @@ def TemporaryFile(
...
+            dir: str = ..., delete: bool =...) -> BinaryIO:
     ...
 def SpooledTemporaryFile(
             max_size: int = ..., mode: str = ..., buffering: int = ...,
             encoding: str = ..., newline: str = ..., suffix: str = ...,
             prefix: str = ..., dir: str = ...) -> BinaryIO:
     ...
 
-class TemporaryDirectory:
+class TemporaryDirectory(ContextManager):
```
Comment (gvanrossum) [inline stdlib/3/tempfile.pyi]: I prefer not to do this -- contextlib.ContextManager is a lie (it doesn't exist in the stdlib, only in the stubs, and we're thinking of moving it to typing), and you don't need to inherit from it in order to be acceptable in a with-statement.

File: stdlib/3/unittest.pyi [hunk 2]
```diff
@@ -8,6 +8,7 @@ from typing import (
 import logging
 import sys
 from types import ModuleType, TracebackType
+from contextlib import ContextManager
```
Comment (gvanrossum) [inline stdlib/3/unittest.pyi]: (Here I'm more OK -- we'll have to fix this once we add ContextManager to typing.py[i] for all PY versions, but that's a ways off.)

**Comment by gvanrossum** [comment]
@FuegoFro Can you push a new version or respond to my comment?

File: stdlib/3/tempfile.pyi (hunk 1 above)
Comment (FuegoFro) [inline stdlib/3/tempfile.pyi]: Gotcha, that makes sense :)

**Comment by FuegoFro** [comment]
Sorry for the delay!

**Pull request merged**